apps/roam/src/utils/initializeObserversAndListeners.ts

```diff
@@ -186,30 +186,52 @@ export const initObservers = async ({
     }
   };
 
+  const customTrigger = onloadArgs.extensionAPI.settings.get(
+    "node-search-trigger",
+  ) as string;
+
   const discourseNodeSearchTriggerListener = (e: Event) => {
     const evt = e as KeyboardEvent;
     const target = evt.target as HTMLElement;
+
     if (document.querySelector(".discourse-node-search-menu")) return;
 
-    if (evt.key === "@" || (evt.key === "2" && evt.shiftKey)) {
-      if (
-        target.tagName === "TEXTAREA" &&
-        target.classList.contains("rm-block-input")
-      ) {
-        const textarea = target as HTMLTextAreaElement;
-        const location = window.roamAlphaAPI.ui.getFocusedBlock();
-        if (!location) return;
-
-        const cursorPos = textarea.selectionStart;
-        const isBeginningOrAfterSpace =
-          cursorPos === 0 ||
-          textarea.value.charAt(cursorPos - 1) === " " ||
-          textarea.value.charAt(cursorPos - 1) === "\n";
-        if (isBeginningOrAfterSpace) {
+    if (
+      target.tagName === "TEXTAREA" &&
+      target.classList.contains("rm-block-input")
+    ) {
+      const textarea = target as HTMLTextAreaElement;
+
+      if (!customTrigger) return;
+
+      const cursorPos = textarea.selectionStart;
+      const textBeforeCursor = textarea.value.substring(0, cursorPos);
+
+      const lastTriggerPos = textBeforeCursor.lastIndexOf(customTrigger);
+
+      if (lastTriggerPos >= 0) {
+        const charBeforeTrigger =
+          lastTriggerPos > 0
+            ? textBeforeCursor.charAt(lastTriggerPos - 1)
+            : null;
+
+        const isValidTriggerPosition =
+          lastTriggerPos === 0 ||
+          charBeforeTrigger === " " ||
+          charBeforeTrigger === "\n";
+
+        const isCursorAfterTrigger =
+          cursorPos === lastTriggerPos + customTrigger.length;
+
+        if (isValidTriggerPosition && isCursorAfterTrigger) {
+          const location = window.roamAlphaAPI.ui.getFocusedBlock();
```

Comment:
Upon further inspection, I'm unsure we actually need this at all.

If it is the case that we are using this as an additional check to be sure the user is actually editing a block (for some edge case that `target.tagName === "TEXTAREA" &&` and `target.classList.contains("rm-block-input")` do not cover, for example) then we should cooerce the value into a boolean (`!!`) and name it as such (eg: `isEditingBlock`).  Even add a comment about the specific use case.  This will help convey the use to other devs.

If it is not needed, we should probably remove it.

I'll approve the PR and leave the decision up to you.